datascience/table.py

```diff
@@ -749,17 +757,22 @@ def wrapped(column):
     return wrapped
 
 
-def _fill_with_zeroes(order, rows, zero=None):
-    """Return a column of the index-1 elements in rows, where the index
-    of each value is determined by matching index-0 to an element of keys.
+def _fill_with_zeroes(keys, rows, zero=None):
+    """Return a column of the index-(-1) elements in rows, where the index
+    of each value is determined by matching index-(0:-1) to an element of
+    keys.
     """
     assert len(rows) > 0
-    index = dict(rows)
+    if not _is_non_string_iterable(keys):
+        keys = [(k,) for k in keys]
```

Comment:
I think everything is being wrapped in a tuple b/c you're assuming `rows` contains tuples and so slicing will give a tuple. Seems fine, but should be documented in the description.